src/components/SingleArticle.jsx

```diff
@@ -22,6 +24,21 @@ const SingleArticle = () => {
       });
   }, [article_id]);
 
+  const handleVote = (voteChange) => {
+    if (!article) return;
+
+    setIsVoting(true);
+    setArticle((prev) => ({ ...prev, votes: prev.votes + voteChange }));
+
+    updateArticleVotes(article_id, voteChange)
+      .then(() => setIsVoting(false))
+      .catch(() => {
+        setIsVoting(false);
+        setArticle((prev) => ({ ...prev, votes: prev.votes - voteChange }));
+        setError("Failed to update vote. Please try again.");
+      });
```

Comment:
Nice logic, but if there is an error for this the whole page disappears and a single error message is displayed, I would recommend making a simple error string appear somewhere next to the voting buttons to say it didnt happen instead of the current behaviour.